cobalt/browser/metrics/cobalt_metrics_service_client_test.cc

```diff
@@ -105,6 +105,15 @@ TEST_F(CobaltMetricsServiceClientTest, TestStubbedSystemFields) {
   ASSERT_EQ(client_->GetAppPackageName(), "");
 }
 
+TEST_F(CobaltMetricsServiceClientTest, UploadIntervalCanBeOverriden) {
+  ASSERT_EQ(client_->GetStandardUploadInterval().InSeconds(), 300);
+  client_->SetUploadInterval(42);
+  ASSERT_EQ(client_->GetStandardUploadInterval().InSeconds(), 42);
+  client_->SetUploadInterval(UINT32_MAX);
+  // UINT32_MAX is the sentinel value to revert back to the default of 5 min.
+  ASSERT_EQ(client_->GetStandardUploadInterval().InSeconds(), 300);
+}
```

Comment:
how about a test to set upload interval to zero? The IDL says it should be ignored.

Reply:
Great call. Added.